python/modules/SoftBDeepAK8SFProducer.py

```diff
@@ -222,6 +373,178 @@ def nGenParts(self, event):
+            numerator_w_up = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged+uncert_w)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged)*topEff_notTagged)).prod()
+            numerator_w_dn = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged-uncert_w)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged)*topEff_notTagged)).prod()
+            numerator_v_up = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged+uncert_bg)*topEff_notTagged)).prod()
+            numerator_v_dn = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged-uncert_bg)*topEff_notTagged)).prod()
+
+            if self.isFastSim:
+                uncert_fast_t  = self.top_fastsferr[fatJetStop0l == 1]
+                uncert_fast_w  = self.top_fastsferr[fatJetStop0l == 2]
+                uncert_fast_bg = self.top_fastsferr[fatJetStop0l == 0]
+                numerator_fast_up = ((topSF_t_tagged+uncert_fast_t)*topEff_t_tagged).prod() * ((topSF_w_tagged+uncert_fast_w)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged+uncert_fast_bg)*topEff_notTagged)).prod()
+                numerator_fast_dn = ((topSF_t_tagged-uncert_fast_t)*topEff_t_tagged).prod() * ((topSF_w_tagged-uncert_fast_w)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged-uncert_fast_bg)*topEff_notTagged)).prod()
+
+                numerator_fast_t_up = ((topSF_t_tagged+uncert_fast_t)*topEff_t_tagged).prod() * ((topSF_w_tagged)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged)*topEff_notTagged)).prod()
+                numerator_fast_t_dn = ((topSF_t_tagged-uncert_fast_t)*topEff_t_tagged).prod() * ((topSF_w_tagged)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged)*topEff_notTagged)).prod()
+                numerator_fast_w_up = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged+uncert_fast_w)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged)*topEff_notTagged)).prod()
+                numerator_fast_w_dn = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged-uncert_fast_w)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged)*topEff_notTagged)).prod()
+                numerator_fast_v_up = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged+uncert_fast_bg)*topEff_notTagged)).prod()
+                numerator_fast_v_dn = ((topSF_t_tagged)*topEff_t_tagged).prod() * ((topSF_w_tagged)*topEff_w_tagged).prod() * (1 - ((topSF_notTagged-uncert_fast_bg)*topEff_notTagged)).prod()
+            else:
+                numerator_fast_up = 0.0
+                numerator_fast_dn = 0.0
+
+                numerator_fast_t_up = 0.0
+                numerator_fast_t_dn = 0.0
+                numerator_fast_w_up = 0.0
+                numerator_fast_w_dn = 0.0
+                numerator_fast_v_up = 0.0
+                numerator_fast_v_dn = 0.0
+
+            self.out.fillBranch("Stop0l_DeepAK8_SFWeight" , numerator/denominator)
```

Comment:
I can't really understand the algo above. Ok, will try to spend sometime on this later on.